www/js/controllers/route-controller.js

```diff
@@ -1,12 +1,24 @@
-angular.module('pvta.controllers').controller('RouteController', function($scope, $stateParams, Route, RouteVehicles, FavoriteRoutes){
+angular.module('pvta.controllers').controller('RouteController', function($scope, $stateParams, Route, RouteVehicles, FavoriteRoutes, Messages){
   var size = 0;
   var route = Route.get({routeId: $stateParams.routeId}, function() {
     route.$save();
     getHeart();
     $scope.stops = route.Stops;
     $scope.vehicles = route.Vehicles
+
+    // Need route to be defined before we can filter messages
+    var messages = Messages.query(function(){
+      var filteredMessages = [];
+      for(var message of messages){
+        if(message.Routes.indexOf($scope.route.RouteId) === -1) { continue; }
```

Comment:
Being super particular here, but changing this to 
```javascript
if(message.Routes.indexOf($scope.route.RouteId) !== -1) filteredMessages.push(message)
``` 
would save us a line and allow the removal of ``continue``, right?

Since ``indexOf`` returns -1 **only** if it doesn't find ``$scope.route.RouteId`` in ``messages``, then the fact that it returns anything else means that the current message applies to the route in question, so it's good to be added.  Please tell me if I'm wrong.